_posts/2020-11-23-boost-your-productivity-with-tailwind-css.md

```diff
@@ -0,0 +1,184 @@
+  style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+  title="JavaScript components | Boost your productivity with Tailwind CSS"
+  allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+  sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+></iframe>
+
+Example 5 is a big step forward visually, but not much is happening Tailwind-wise. To achieve this popup style, we only had to add one new configuration option in `tailwind.config.js`. The rest is just JSX and React. As a matter of fact, in Tailwind v2, we wouldn't even have needed to make changes to the configuration.
+
+As your component grows and you have to render content based on props using the ternary operator (`myProp ? ... : ...`) quickly becomes messy with Tailwind. I would recommend something along the lines of [classnames](https://github.com/JedWatson/classnames) to handle your classes.  Also, you should always use the full name of the class instead of concatenating them. So instead of writing `` `text-${myVar ? 'xl' : 'base'}` `` write it like this `myVar ? 'text-xl' : 'text-base'`. This rule applies even when using other tools. The ternary operator is just an example. Otherwise, PurgeCSS may remove the classes during purging.
+
+## Conclusion
+
+Tailwind and utility-based CSS, in general, is a somewhat controversial topic among the front-end community. To me, it has completely changed how I feel about styling web applications. It feels very intuitive, albeit like with many technologies, it has its hurdles. Despite that, I think the pros outweigh the cons.
+
+Pros
+
+* No energy spent on naming classes
+* Everything is customizable
+* It's easy and precise to modify how your UI component looks like
+* No CSS to maintain
+
+Cons
+
+* Something feels off with elements and components with bloated classes
+* Extracting sensible components can be difficult
+* No ready-made UI components (although Tailwind isn't a CSS framework, so to speak, and some projects aim to solve that)
+
+![This is brilliant, but I like utility](/img/boost-your-productivity-with-tailwind-css/but-i-like-utility.png)
+
+This blog post doesn't cover everything Tailwind has to offer, and a couple of fun ones that you can explore on your own are animations and plugins. Lastly, I want to give you a headstart and leave you off with some helpful links:
+
+* [Tailwind CSS IntelliSense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) for VS Code
+* [Headwind](https://marketplace.visualstudio.com/items?itemName=heybourn.headwind), Tailwind CSS formatter for VS Code
+* [A curated list of awesome tools, material, and examples](https://github.com/aniftyco/awesome-tailwindcss)
```

Comment:
This is a good conclusion. I was wondering, since components came up, which framework have you been mixing this up, or have the most experience? Bringing in some extra comment from viewpoint of mixing this up with Vue/React in professional work, like componentizing the UI, might be great way to end this.